ui-admin/src/portal/languages/PortalEnvLanguageEditor.tsx

```diff
@@ -0,0 +1,230 @@
+}, {
+  'languageName': 'Türk',
+  'languageCode': 'tr'
+}]
+
+/**
+ * Table which allows viewing, deleting, and creating new answer mappings.
+ */
+export default function PortalEnvLanguageEditor({ items, setItems, readonly } : {
+    items: PortalEnvironmentLanguage[],
+    setItems: (items: PortalEnvironmentLanguage[]) => void,
+  readonly: boolean
+  }
+) {
+  const [itemSelectedForDeletion, setItemSelectedForDeletion] = useState<PortalEnvironmentLanguage | null>(null)
+  // state for new item
+  const [newItem, setNewItem] =
+    useState<EditablePortalEnvironmentLanguage>(makeEmptyNewItem())
+  const deleteItem = async () => {
+    if (!itemSelectedForDeletion) {
+      return
+    }
+
+    const filteredItems = items.filter(m => m.languageCode !== itemSelectedForDeletion.languageCode)
+    setItems(filteredItems)
+    setItemSelectedForDeletion(null)
+  }
+
+  const addNewItem = (item: PortalEnvironmentLanguage) => {
+    const newItems = [...items, item]
+    setNewItem(makeEmptyNewItem())
+    setItems(newItems)
+  }
+
+  const updateItem = (item: PortalEnvironmentLanguage) => {
+    if (item.languageCode === newItem.languageCode) {
+      addNewItem(item)
+      return
+    }
+    const newItems = items.map(m => m.languageCode === item.languageCode ? item : m)
+    setItems(newItems)
+  }
+
+
+  const columns: ColumnDef<PortalEnvironmentLanguage>[] = useMemo(() => {
+    const baseCols: ColumnDef<PortalEnvironmentLanguage>[] = [{
+      header: 'Name',
+      accessorKey: 'languageName',
+      cell: ({ row }) => {
+        const value = row.original.languageName
+        if (isEditable(row.original)) {
+          return row.original.isEditing && <Creatable
+            aria-label={'Language name'}
+            options={LANGUAGE_OPTIONS.map(name => {
+              return {
+                value: name.languageName,
+                label: name.languageName
+              }
+            })}
+            value={row.original.languageName && {
+              value: row.original.languageName,
+              label: row.original.languageName
+            }}
+            onChange={e => e && setNewItem(
+              {
+                ...row.original,
+                languageName: e.value,
+                languageCode: LANGUAGE_OPTIONS.find(lang => lang.languageName === e.value)?.languageCode || '',
+                isEditing: true
+              })}
+          />
+        }
+        return value
+      }
+    }, {
+      header: 'Code',
+      accessorKey: 'languageCode',
+      cell: ({ row }) => {
+        const value = row.original.languageCode
+        if (isEditable(row.original)) {
+          return row.original.isEditing && <input
+            type="text"
+            aria-label={'Language code'}
+            onChange={e => setNewItem({
+              ...row.original,
+              languageName: e.target.value,
+              isEditing: true
+            })}
+            value={row.original.languageCode}
+          />
```

Comment:
Nit: needs styling.

Also, editing this didn't work for me; I think it should set `languageCode` instead of `languageName`.